### PR TITLE
[docs] Add a changelog for v1.0.0

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**
\cc @DevExpress/testcafe-docs @miherlosev 

The `Breaking Changes` section is copied from https://github.com/DevExpress/testcafe/pull/3388 - you may not check it. And it's `WIP` because of that section - I hope we can add some positive moments to it.

You are welcome to check the rest!

### CODE REVIEW

**Comment by VasilyStrelyaev** [comment]
**Note** that it's PRed against `master` not `new-docs` so let's merge it when we release.

Just check now pls.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,198 @@
...
+
+The `--disable-test-syntax-validation` command line flag and the `disableTestSyntaxValidation` option for the [runner.run](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#run) API method that disabled test syntax validation were removed in v1.0.0.
+
+#### :boom: Custom Request Hooks: Asynchronous API
+
+[Request hook](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/) methods became asynchronous in TestCafe v1.0.0.
+
+If you use a [custom request hook](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html), add the `async` keyword before the [onRequest](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onrequest-method) and [onResponse](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onresponse-method) methods.
```
Comment (kirovboris) [inline CHANGELOG.md]: It's not obligatory, you can still use it without `async`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,198 @@
...
+import { RequestHook } from 'testcafe';
+class MyRequestHook extends RequestHook {
+    constructor (requestFilterRules, responseEventConfigureOpts) {
+        super(requestFilterRules, responseEventConfigureOpts);
+        // ...
+    }
+    async onRequest (event) {
+        // ...
+    }
+    async onResponse (event) {
```
Comment (kirovboris) [inline CHANGELOG.md]: Please add an empty line.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,198 @@
...
+* [reportTaskStart](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reporttaskstart)
+* [reportFixtureStart](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reportfixturestart)
+* [reportTestDone](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reporttestdone)
+* [reportTaskDone](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reporttaskdone)
+
+```js
+async reportTaskStart (startTime, userAgents, testCount) {
+    // ...
+},
+async reportFixtureStart (name, path, meta) {
```
Comment (kirovboris) [inline CHANGELOG.md]: Please add empty lines between methods.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,198 @@
...
+        // ...
+    }
+}
+```
+
+#### :boom: Custom Reporter Plugins: Asynchronous API
+
+TestCafe v1.0.0 also introduces asynchronous API for [reporter plugins](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/).
+
+If you maintain a custom reporter plugin, add the `async` keyword before each reporter method:
```
Comment (kirovboris) [inline CHANGELOG.md]: It's not obligatory as well.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,198 @@
...
+},
+async reportTestDone (name, testRunInfo, meta) {
+    // ...
+},
+async reportTaskDone (endTime, passed, warnings, result) {
+    // ...
+}
+```
+
+#### :boom: Programming Interface: Multiple Method Calls Prohibited
```
Comment (kirovboris) [inline CHANGELOG.md]: It needs some motivation IMO.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,198 @@
...
+```js
+runner
+    .src(['/home/user/tests/fixture1.js', 'fixture5.js'])
+    .browsers(['chrome', 'firefox:headless'])
+    .reporter(['minimal', { name: 'json', output: 'report.json' }]);
+```
+
+### Enhancements
+
+#### :gear: Video Recording [#2151](https://github.com/DevExpress/testcafe/issues/2151)
```
Comment (kirovboris) [inline CHANGELOG.md]: We should mention supported browsers here.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,198 @@
 # Changelog
 
+## v1.0.0 (2018-2-7)
```
Comment (kirovboris) [inline CHANGELOG.md]: We should add a couple words about
https://github.com/DevExpress/testcafe/pull/3356
https://github.com/DevExpress/testcafe/pull/3177
Also I'm not sure we should put the breaking changes before new features.

File: CHANGELOG.md (hunk 5 above)
Comment (VasilyStrelyaev) [reply CHANGELOG.md]: I'll copy motivation from here once it's approved:
https://github.com/DevExpress/testcafe/pull/3388/files#diff-1ce235e72b3786322b3f4b017ce27cd7R80

Will it be enough?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,289 @@
...
+    {
+        "videoPath": "artifacts/videos/"
+    }
+    ```
+
+TestCafe records all tests and saves each recording in a separate file. You can change this behavior in [video options](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/screenshots-and-videos.html#basic-video-options). You can also customize [video encoding parameters](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/screenshots-and-videos.html#video-encoding-options).
+
+#### :gear: Configuration File ([#3131](https://github.com/DevExpress/testcafe/issues/3131))
+
+TestCafe now allows you to store its settings in the `.testcaferc.json` [configuration file](https://devexpress.github.io/testcafe/documentation/using-testcafe/configuration-file.html) (with support for `JSON5` syntax).
```
Comment (miherlosev) [inline CHANGELOG.md]: Add link to https://json5.org/

**Pull request merged**